hazelcast-client/src/main/java/com/hazelcast/client/map/impl/nearcache/ClientHeapNearCache.java

```diff
@@ -260,4 +196,71 @@ public NearCacheStatsImpl getNearCacheStats() {
         stats.setOwnedEntryMemoryCost(ownedEntryMemoryCost);
         return stats;
     }
+
+    private void fireCacheEviction() {
+        if (canEvict.compareAndSet(true, false)) {
+            try {
+                executionService.executeInternal(new Runnable() {
+                    public void run() {
+                        try {
+                            Set<NearCacheRecord> records = new TreeSet<NearCacheRecord>(selectedComparator);
+                            records.addAll(cache.values());
+                            int evictSize = (int) (cache.size() * EVICTION_FACTOR);
+                            int i = 0;
+                            for (NearCacheRecord record : records) {
+                                cache.remove(record.getKey());
+                                stats.incrementEvictions();
+                                if (++i > evictSize) {
```

Comment:
I'm already prototyping to replace those classes with `DefaultNearCache`. So I think it's okay to stick with it.